api/types/types.go

```diff
@@ -190,3 +190,42 @@ type Workflow struct {
 	Parameters    []WorkflowParam `yaml:"parameters"`
 	Source        string          `valid:"isValidWorkflow" yaml:"source"`
 }
+
+// Node describes a node
+type NodeRule struct {
+	UUID     string
+	Type     string `valid:"nonzero"`
+	Name     string `valid:"nonzero"`
+	Metadata string
+	Action   string `valid:"regexp=^(create|update)$"`
+	Query    string
```

Comment:
`valid:"isGremlinExpr"

Reply:
this is optional field when creating node rule. so if we validate it will fail when it is empty.